new/subscribes-block.md

```diff
@@ -0,0 +1,73 @@
+---
+RFC: unassigned
+Author: John Keiser <jkeiser@chef.io>
+Status: Draft
+Type: Standards Track
+Tracking:
+  - https://github.com/chef/chef/issues/X
+---
+
+# Subscribe Blocks
+
+Add the ability to run a block instead of an action in subscribes and notifies.
+
+## Motivation
+
+    As a Chef user,
+    I want a pattern allowing me to avoid the service restart problem,
+    So that my services don't get restarted the first time I run a new machine.
+
+    As a Chef user,
+    I want a pattern allowing me to accumulate desired state in one resource before it is executed,
+    So that I can let many different resources update at once in a batch.
+
+    As a Chef user,
+    I want update patterns that don't actually converge anything directly,
+    So that I can accomplish these patterns without showing a confusing new action in the Chef log.
+
+## Specification
+
+notifies and subscribes each have a new form, taking a block instead of an action, which will call the block when the notification occurs rather than running an action. The block is evaluated in the context of the subscribing resource.
+
+The old forms all still work; the new form is additive.
+
+## Uses
+
+### Accumulating desired state
+
+```ruby
+# When the templates with the list of packages changes, make sure and install the packages.
+template '/my-packages.txt' do
+  notifies('package[all_the_things]', :immediately) { packages.push *current_value.content.split(",") }
+end
+template '/chef-packages.txt' do
+  notifies('package[all_the_things]', :immediately) { packages.push *current_value.content.split(",") }
+end
+package 'all_the_things' do
+  package_name []
+end
+```
+
+### Avoiding multiple destructive operations
+
+```ruby
+file '/etc/config-file.txt' do
+  content 'config!'
+end
+file '/etc/init.d/x/start.sh' do
+end
+# The service will only be processed once; if the configuration is updated, we
+# upgrade the action to :reload; if the configuration is started, we upgrade
+# the action to :restart.
```

Comment:
What decides that reload gets upgraded to restart? 

Is it the order of the subscribes entries in the resource below? Seems magic. You could end up with something similar to "before" in #154 if that was true:
```
service 'x' do
  subscribes('execute[do we need to make a change]', :immediately) { action :stop }
  subscribes('execute[make the change]', :immediately) { action :start }
do
```

Or are we putting that in the code? If that, can we discuss and possibly document that matrix of possibilities?